modules/standard/Sets.chpl

```diff
@@ -0,0 +1,496 @@
+
+      :arg other: The set to compare against.
+      :return: Whether or not this set is a proper subset of `other`.
+    */
+    proc const properSubset(const ref other: set(eltType, ?)): bool {
+      return false;
+    }
+
+    /*
+      Returns `true` if the value of every element in the set `other` is
+      contained in this set.
+
+      :arg other: The set to compare against.
+      :return: Whether or not this set is a superset of `other`.
+    */
+    proc const superset(const ref other: set(eltType, ?)): bool {
+      return false;
+    }
+
+    /*
+      Returns `true` if this set is a proper superset of `other`. That is,
+      the value of every element in `other` is contained in this set, and
+      this set does not equal `other`.
+
+      :arg other: The set to compare against.
+      :return: Whether or not this set is a proper superset of `other`.
+    */
+    proc const properSuperset(const ref other: set(eltType, ?)): bool {
+      return false;
+    }
+
+    /*
+      Remove the item from this set with a value equal to `x`.
+
+      .. warning::
+
+        Removing an element from this set may invalidate existing references
+        to the elements contained in this set.
+    
+      :arg x: The element to remove.
+
+      :throws IllegalArgumentError: If the list contains no such element.
+    */
+    proc remove(const ref x: eltType): eltType throws {
+      return;
+    }
+
+    /*
+      Remove the item `x` if it is present in this set, but do nothing if it
+      is not.
+
+      .. warning::
+
+        Discarding an element from this set may invalidate existing references
+        to the elements contained in this set.
+
+      :arg x: The element to discard.
+    */
+    proc discard(const ref x: eltType) {
+      return;
+    }
+
+    /*
+      Clear the contents of this set.
+
+      .. warning::
+
+        Clearing the contents of this set will invalidate all existing
+        references to the elements contained in this set.
+    */
+    proc clear() {
+      return;
+    }
+
+    /*
+      Iterate over the elements of this set.
+
+      :yields: A reference to one of the elements contained in this set.
+    */
+    iter these() ref {
+      return nil;
+    }
+
+    /*
+      Write the contents of this set to a channel.
+
+      :arg ch: A channel to write to.
+    */
+    proc const writeThis(ch: channel) {
+      return;
+    }
+
+    /*
+      Returns `true` if this set contains zero elements.
+
+      :return: `true` if this set is empty.
+      :rtype: `bool`
+    */
+    inline proc const isEmpty(): bool {
+      return false;
+    }
+
+    /*
+      The current number of elements contained in this set.
+    */
+    inline proc const size {
+      return 0;
+    }
+
+    /*
+      Returns a new DefaultRectangular array containing a copy of each of the
+      elements contained in this set. The returned array is not guaranteed
+      to follow any particular ordering.
```

Comment:
I'd make `toArray` sort the set, personally. Is there a reason you didn't do that? To sort or not could be an option argument.

Reply:
I stated "no particular ordering" because an implementation of `set` may not maintain its elements internally in any particular order.

It would be trivial to remove that description and then sort the output before returning it, though.

Reply:
Yes, that's what I'm talking about doing

Reply:
Ok, that seems reasonable.